File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
// class Solution {
// public:
//     bool hasDuplicate(vector<int>& nums) {
        
//     }
// };

class Solution{
public: bool hasDuplicate(vector<int>& nums){

    int n = nums.size();
    unordered_set <int> uset;
    for(int i=0; i < n; i++){

        if (uset.count(nums[i])){
            return true;
        }

        uset.insert(nums[i]);

    }

    return false;

}
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        

        l =0
        r = len(nums)-1
        ans = 0
        while l<=r:
            
            mid = (l+r)//2

            ans = nums[mid]

            print(ans)

            if ans > nums[r]:
                l = mid+1
            elif ans < nums[r]:
                r = mid
            else:
                return ans

        return ans


                



            